flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/sink/writer/AsyncSinkWriter.java

```diff
@@ -290,17 +290,33 @@ private void registerCallback() {
     @Override
     public void write(InputT element, Context context) throws IOException, InterruptedException {
         while (bufferedRequestEntries.size() >= maxBufferedRequests) {
-            mailboxExecutor.tryYield();
+            flush();
```

Comment:
2. I have changed the tryYield to a flush. Consider the case where the number of requests in the buffer is greater than the `maxBufferedRequests`. Calling tryYield is not guaranteed to reduce the number of elements in the buffer, e.g. if there are no more mailbox messages scheduled to be run or ready to run.